### PR TITLE
urequests: support ports other than default

### PR DESCRIPTION
**Pull request by garybake**
Sourced from https://github.com/kennethreitz/requests/blob/190cddd4ae7d649ea6d85d062fd6e4a69d0cf417/requests/packages/urllib3/util/url.py#L179

### CODE REVIEW

File: urequests/urequests.py [hunk 1]
```diff
@@ -32,12 +32,21 @@ def json(self):
...
         path = ""
     if proto != "http:":
         raise ValueError("Unsupported protocol: " + proto)
+    
+    if ':' in hostname:
+        host, port = hostname.split(':', 1)
+        if not port.isdigit():
+            raise ValueError('Unable to parse port')
+        port = int(port)
+    else:
```
Comment (pfalcon) [inline urequests/urequests.py]: Please avoid this if, and having both "host" and "hostname" vars.

File: urequests/urequests.py [hunk 2]
```diff
@@ -32,12 +32,21 @@ def json(self):
...
     except ValueError:
-        proto, dummy, host = url.split("/", 2)
+        proto, dummy, hostname = url.split("/", 2)
         path = ""
     if proto != "http:":
         raise ValueError("Unsupported protocol: " + proto)
+    
+    if ':' in hostname:
+        host, port = hostname.split(':', 1)
+        if not port.isdigit():
```
Comment (pfalcon) [inline urequests/urequests.py]: Please avoid this if, int("foo") already throws ValueError.

**Comment by garybake** [comment]
Thank you for the comments. 
For my own curiosity - the removal of the` if ':' in hostname:` was this due to efficiency with micropython or just a more pythonic way of doing it?

**Comment by pfalcon** [comment]
> Please avoid this if, and having both "host" and "hostname" vars.

I'm sorry for formulating it in such confusing way. I meant "avoid else". I.e. instead of:

~~~
if ...:
    var = expr
else:
    var = const
~~~

preferrable to use:

~~~
var = const
if ...:
    var = expr
~~~

This will make code a tiny bit shorter and tiny bit more efficient.

**Comment by garybake** [comment]
Gotcha - I've updated the file

File: urequests/urequests.py [hunk 3]
```diff
@@ -39,7 +39,13 @@ def request(method, url, data=None, json=None, headers={}, stream=None):
     if proto != "http:":
         raise ValueError("Unsupported protocol: " + proto)
 
-    ai = usocket.getaddrinfo(host, 80)
+    port = 80
+    if ':' in host:
+        _host, port = host.split(':', 1)
+        host = _host
```
Comment (pfalcon) [inline urequests/urequests.py]: This is still superfluous. MicroPython allocates a memory slot for each local variable, more variables == more memory used. So, if it's possible to reuse a variable, that should be done - on the level of micropython-lib for sure. I'm fixing this myself.

**Comment by pfalcon** [comment]
Merged with fixes, thanks.

**Pull request closed without merging**